tests/DwitTech.AccountService.Core.Tests/Services/UserServiceTest.cs

```diff
@@ -285,5 +286,60 @@ public async Task Logout_Should_Return_True_If_User_Was_Successfully_Logged_Out(
                 throw new Exception($"{ex.Message}");
             }
         }
+
+        [Fact]
+        public async Task DeleteUserAsync_DeletesUserWithValidId()
+        {
+            // Arrange
+            int userIdToDelete = 1;
+            var iUserRepoMock = new Mock<IUserRepository>();
+            var iLoggerMock = new Mock<ILogger<UserService>>();
+            var iRoleRepoMock = new Mock<IRoleRepository>();
+            var mockAuthRepository = new Mock<IAuthenticationRepository>();
+            var iActivationServiceMock = new Mock<IActivationService>();
+            var iAuthenticationService = new Mock<IAuthenticationService>();
+            var iEmailServiceMock = new Mock<IEmailService>();
+            var iConfigurationMock = new Mock<IConfiguration>();
+            var mockHttpContextAccessor = new Mock<IHttpContextAccessor>();
+
+
+            var userService = new UserService(iUserRepoMock.Object, iRoleRepoMock.Object, mockAuthRepository.Object, iLoggerMock.Object,
+                iActivationServiceMock.Object, iEmailServiceMock.Object, iConfigurationMock.Object, iAuthenticationService.Object, mockHttpContextAccessor.Object);
+
+            iUserRepoMock.Setup(repo => repo.DeleteUserAsync(userIdToDelete))
+                .Returns(Task.CompletedTask);
+
+            // Act
+            await userService.DeleteUserAsync(userIdToDelete);
+
+            // Assert
+            iUserRepoMock.Verify(repo => repo.DeleteUserAsync(userIdToDelete), Times.Once);
+        }
+
+        [Fact]
+        public async Task DeleteUserAsync_ThrowsDbUpdateException_WhenDatabaseUpdateFails()
```

Comment:
I think this test should be in your repo test, not here.

Reply:
> I think this test should be in your repo test, not here.

Did you check my DeleteUserAsync implementation in my user service? I did a catch for DbUpdateException, hence the test.